vault/src/vaults.rs

```diff
@@ -136,9 +156,10 @@ impl<P: RelayPallet + BtcRelayPallet, B: BitcoinCoreApi + Clone> VaultTheftMonit
             bitcoin::stream_in_chain_transactions(self.bitcoin_core.clone(), self.btc_height, num_confirmations).await;
 
         while let Some(Ok((block_hash, tx))) = stream.next().await {
-            if let Err(err) = self.check_transaction(tx, block_hash, num_confirmations).await {
+            if let Err(err) = self.check_transaction(tx.clone(), block_hash, num_confirmations).await {
                 tracing::error!("Failed to check transaction: {}", err);
             }
+            self.monitor_bitcoin_metrics(tx, num_confirmations).await;
```

Comment:
This doesn't make a lot of sense to me - why do we listen on the bitcoin chain for our own transactions? Why not just call `update_bitcoin_metrics` directly after making a transaction?

Reply:
I'm thinking that not all bitcoin transactions may be made through this client. What if the client crashes and a redeem payment is made manually?